developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -2114,7 +2114,38 @@ where `my_event.foreman` is the name of the event you want to subscribe to. You
   subscribe /.foreman$/, MyPlugin::MySubscriber
 ....
 
-You can find all observable events by calling `Foreman::EventSubscribers.all_observable_events` in the Rails console. Keep in mind that this list only contains events that were set by the `set_hook` method.
+Example events emitted by creating, updating or deleting of selected records (subclasses of `ApplicationRecord` which are defined via `set_hook` method):
+
+* `subnet_created.event.foreman`
+* `subnet_updated.event.foreman`
+* `subnet_destroyed.event.foreman`
+
+Payload for records is the record model object itself under key `object` and `context` with additional logging context. Keep in mind that the model classes are not subject of stable API, they will change in the future. It's recommended not to publish full object but to strip down exposed information to bare minimum (e.g. host name and ID). 
+
+Example events emitted by performing background jobs (subclasses of `ApplicationJob`):
+
+* `template_render_job_performed.event.foreman`
+* `create_rss_notifications_performed.event.foreman`
+
+Payload for background jobs is the serialized active job hash (see `ActiveJob#serialize` method) named `job` and `context` with additional logging context. Example:
+
+```
+{"job" =>
+  {"job_class"=>"DoNothingJob",
+   "job_id"=>"377c30b8-25ed-4c9f-a03f-028132029324",
+   "provider_job_id"=>nil,
+   "queue_name"=>"default",
+   "priority"=>nil,
+   "arguments"=>["one"],
+   "executions"=>1,
+   "exception_executions"=>{},
+   "locale"=>"en",
+   "timezone"=>"UTC",
+   "enqueued_at"=>"2020-11-03T15:33:31Z"},
+ "context" => {"user_login"=>"secret_admin", "user_admin"=>true}}
+```
+
+You can find all observable events by calling `Foreman::EventSubscribers.all_observable_events` in the Rails console. 
```

Comment:
The job events don't show up in the list returned by `Foreman::EventSubscribers.all_observable_events`. From what I've seen from the implementation I'd say it should be doable.

Either it should really return *all* events or set the expectations straight by not including all in the name of the method

Reply:
This is a bug, the method should return all events which are available, this is an overlook. Will fix.

> If we want to eventually have an UI where the user could select from a list of events, then I'd lean towards the first option.

We already have a UI where user picks from dropdown showing (currently whitelisted) events, yeah. That is the plan.